### DISCUSSION TITLE
v2.0.0-beta.27 svelte + dev

### DISCUSSION THREAD

**Discussion by CharlieGo19** [wailsapp/wails#1054] [Releases]
Hi, I ❤️ wails and really happy there's a svelte template for V2.

I am having trouble with getting changes to frontend to reflect in browser&app, I'm having to stop and restart wails dev each time I make a change to the project.

I tried using -reloaddirs ./frontend to see if that makes any difference, but with little success.

Am I missing something, frontend frameworks are not my arena so I'm a fish out of water.

Thanks for the support,

Charlie.

**_edit: Scrap that last edit 😛. OS: macOS Monterey 12.1_**

**Comment by leaanthony** [comment]
Hey Charlie! Glad you're enjoying it. There's been some work on the Dev setup so I'll test and see if I need to update the default template. Thanks for your patience 🙏

**Comment by alanchenchen** [comment]
Has this issue been resolved? I have the same problem.
```bash
$ wails version
v2.0.0-beta.30

system
------
OS:             Windows 10 Pro
Version:        1909 (Build: 18363)
ID:
Go Version:     go1.17.6
Platform:       windows
Architecture:   amd64
```
And i did some tries but not works.
eg:

1. cd ./frontend, input `npm run dev` in terminal, `http://localhost:5000` will be access for local frontend assets server
2. change `devserverurl` to `http://localhost:5000` in wails.json
3. the wails apps load http://localhost:5000 successfully although go bindings does not work

**Comment by BillBuilt** [comment]
@alanchenchen  Are you using Svelte or SvelteKit? If using Svelte (not SvelteKit), are you using Vite or another bundler? The biggest issue I saw with Vite is how it works out-of-the-box. VIte's dev mode, by default, runs off of the un-built raw code and somehow "builds on the fly as needed". The problem I ran into with Vite and Wails, is that Wails expects (and most likely needs) that build process for dev mode. Therefore I switched to using Vite’s build command with the `—watch` flag instead of its standard dev command. This works well with Wails, since Vite only builds into `/dist` and then watches for changes and rebuilds when needed, but does not launch a server.

I believe the issues with Wails and SvelteKit is that, as far as I know, you cannot pass that `—watch` flag to Vite. In fact, with SvelteKit, all commands are passed through to Vite via its own commands.

If you are using Svelte with Vite, I'd not use the `devserverurl` option and change your `npm run dev` script in your Svelte `package.json` from just `vite` to `vite build --watch` and give that a try. There will still be times when things seem to get "disconnected", for lack of a better description, but it's not that often and is getting better with each release.

If you're using another bundler, mayby see if there's an option that would work in a similar way.
